[Qase][16:25:09][debug] Config: {
    "debug": true,
    "environment": "local",
    "exclude_params": [],
    "execution_plan": {
        "path": "./build/qase-execution-plan.json"
    },
    "fallback": "report",
    "framework": {
        "playwright": {
            "output_dir": "test-results",
            "trace": "off",
            "video": "off"
        },
        "pytest": {
            "capture_logs": true,
            "xfail_status": {
                "xfail": "skipped",
                "xpass": "passed"
            }
        }
    },
    "mode": "testops",
    "profilers": [
        "network",
        "sleep"
    ],
    "report": {
        "connection": {
            "format": "json",
            "path": "./build/qase-report"
        },
        "driver": "local"
    },
    "testops": {
        "api": {
            "host": "qase.io",
            "token": "bf98087c2b769f39a4741acf720a27171d3128ec64fa2a5bc4987e7425834825"
        },
        "batch": {
            "size": 100
        },
        "defect": false,
        "plan": {},
        "project": "RANDOM",
        "run": {
            "complete": true,
            "tags": [],
            "title": "pytestrun"
        }
    }
}
[Qase][16:25:10][debug] Host data: {'system': 'darwin', 'machineName': 'Ankitas-MacBook-Air.local', 'release': '24.6.0', 'version': '15.6', 'arch': 'arm64', 'python': '3.13.5', 'pip': '24.3.1', 'framework': '8.3.4', 'reporter': '6.3.3', 'commons': '3.5.0', 'apiClientV1': '1.2.1', 'apiClientV2': '1.2.1'}
[Qase][16:25:10][debug] Preparing API client
[Qase][16:25:10][debug] API client prepared
[Qase][16:25:10][debug] Preparing API V2 client
[Qase][16:25:10][debug] API V2 client prepared
[Qase][16:25:10][debug] Getting environment local
[Qase][16:25:11][debug] Environment local not found
[Qase][16:25:11][debug] Getting project RANDOM
[Qase][16:25:11][debug] Project RANDOM found: {"title": "Random", "code": "RANDOM", "counts": {"cases": 7, "suites": 5, "milestones": 0, "runs": {"total": 47, "active": 14}, "defects": {"total": 3, "open": 3}}}
[Qase][16:25:11][debug] Starting run
[Qase][16:25:11][debug] Creating test run with parameters: {'title': 'pytestrun', 'description': 'Automated Run 2025-08-12 16:25:11.431593', 'environment_id': None, 'plan_id': None, 'is_autotest': True, 'start_time': '2025-08-12 10:55:11', 'tags': []}
[Qase][16:25:12][debug] Run ID: 76
[Qase][16:25:15][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 2456,
        "end_time": 1754996115.133686,
        "stacktrace": null,
        "start_time": 1754996112.6774888,
        "status": "passed",
        "thread": "34136-MainThread"
    },
    "fields": {},
    "id": "1a52f407-240d-4554-a0a5-4960c696b736",
    "message": null,
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_has_title[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_has_title"
}
[Qase][16:25:15][debug] Result test_has_title added
[Qase][16:25:20][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 5622,
        "end_time": 1754996120.756883,
        "stacktrace": "page = <Page url='https://playwright.dev/docs/intro'>\n\n    def test_get_started_link(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: typo in heading name\n        page.get_by_role(\"link\", name=\"Get started\").click()\n>       expect(page.get_by_role(\"heading\", name=\"Instalation\")).to_be_visible()  # typo\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 5000ms\nE         -   - waiting for get_by_role(\"heading\", name=\"Instalation\")\n\ntests/test_example.py:15: AssertionError",
        "start_time": 1754996115.13417,
        "status": "failed",
        "thread": "34136-MainThread"
    },
    "fields": {},
    "id": "3ce43b3a-87b8-42a3-ae77-9adb0d15b49f",
    "message": "AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 5000ms\n  -   - waiting for get_by_role(\"heading\", name=\"Instalation\")",
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_get_started_link[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_get_started_link"
}
[Qase][16:25:20][info] See why this test failed: https://app.qase.io/run/RANDOM/dashboard/76?source=logs&status=%5B2%5D&search=test_get_started_link
[Qase][16:25:20][debug] Result test_get_started_link added
[Qase][16:25:26][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 5498,
        "end_time": 1754996126.2565382,
        "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_wrong_url(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: wrong URL assertion\n>       expect(page).to_have_url(\"https://example.com\")  # wrong URL\nE       AssertionError: Page URL expected to be 'https://example.com'\nE       Actual value: https://playwright.dev/ \nE       Call log:\nE         - PageAssertions.to_have_url with timeout 5000ms\nE         -   - waiting for locator(\":root\")\nE         -     9 \u00d7 locator resolved to <html lang=\"en\" dir=\"ltr\" data-theme=\"light\" data-has-hydrated=\"true\" data-theme-choice=\"system\" class=\"plugin-pages plugin-id-default\" data-rh=\"lang,dir,class,data-has-hydrated\">\u2026</html>\nE         -       - unexpected value \"https://playwright.dev/\"\n\ntests/test_example.py:21: AssertionError",
        "start_time": 1754996120.7584739,
        "status": "failed",
        "thread": "34136-MainThread"
    },
    "fields": {},
    "id": "c4b68e37-c9be-4eb6-9fab-2a07ce46a69e",
    "message": "AssertionError: Page URL expected to be 'https://example.com'\nActual value: https://playwright.dev/ \nCall log:\n  - PageAssertions.to_have_url with timeout 5000ms\n  -   - waiting for locator(\":root\")\n  -     9 \u00d7 locator resolved to <html lang=\"en\" dir=\"ltr\" data-theme=\"light\" data-has-hydrated=\"true\" data-theme-choice=\"system\" class=\"plugin-pages plugin-id-default\" data-rh=\"lang,dir,class,data-has-hydrated\">\u2026</html>\n  -       - unexpected value \"https://playwright.dev/\"",
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_fail_wrong_url[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_fail_wrong_url"
}
[Qase][16:25:26][info] See why this test failed: https://app.qase.io/run/RANDOM/dashboard/76?source=logs&status=%5B2%5D&search=test_fail_wrong_url
[Qase][16:25:26][debug] Result test_fail_wrong_url added
[Qase][16:25:31][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 5549,
        "end_time": 1754996131.806358,
        "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_element_not_found(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: element that doesn't exist\n>       expect(page.get_by_text(\"Nonexistent Text\")).to_be_visible()\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 5000ms\nE         -   - waiting for get_by_text(\"Nonexistent Text\")\n\ntests/test_example.py:27: AssertionError",
        "start_time": 1754996126.2573519,
        "status": "failed",
        "thread": "34136-MainThread"
    },
    "fields": {},
    "id": "2204994a-4149-4e1c-b665-1f5fffa72ffc",
    "message": "AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 5000ms\n  -   - waiting for get_by_text(\"Nonexistent Text\")",
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_fail_element_not_found[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_fail_element_not_found"
}
[Qase][16:25:31][info] See why this test failed: https://app.qase.io/run/RANDOM/dashboard/76?source=logs&status=%5B2%5D&search=test_fail_element_not_found
[Qase][16:25:31][debug] Result test_fail_element_not_found added
[Qase][16:25:37][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 5574,
        "end_time": 1754996137.3817348,
        "stacktrace": "page = <Page url='https://playwright.dev/docs/intro'>\n\n    def test_fail_incorrect_text(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        page.get_by_role(\"link\", name=\"Get started\").click()\n        # Fail: text is not matching\n>       expect(page.get_by_role(\"heading\", name=\"Installation\")).to_have_text(\"Wrong Text\")\nE       AssertionError: Locator expected to have text 'Wrong Text'\nE       Actual value: Installation \nE       Call log:\nE         - LocatorAssertions.to_have_text with timeout 5000ms\nE         -   - waiting for get_by_role(\"heading\", name=\"Installation\")\nE         -     6 \u00d7 locator resolved to <h1>Installation</h1>\nE         -       - unexpected value \"Installation\"\n\ntests/test_example.py:34: AssertionError",
        "start_time": 1754996131.807461,
        "status": "failed",
        "thread": "34136-MainThread"
    },
    "fields": {},
    "id": "f69ec1c1-cddc-4486-9362-cb48ed295775",
    "message": "AssertionError: Locator expected to have text 'Wrong Text'\nActual value: Installation \nCall log:\n  - LocatorAssertions.to_have_text with timeout 5000ms\n  -   - waiting for get_by_role(\"heading\", name=\"Installation\")\n  -     6 \u00d7 locator resolved to <h1>Installation</h1>\n  -       - unexpected value \"Installation\"",
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_fail_incorrect_text[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_fail_incorrect_text"
}
[Qase][16:25:37][info] See why this test failed: https://app.qase.io/run/RANDOM/dashboard/76?source=logs&status=%5B2%5D&search=test_fail_incorrect_text
[Qase][16:25:37][debug] Result test_fail_incorrect_text added
[Qase][16:25:40][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 3535,
        "end_time": 1754996140.9187148,
        "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_timeout(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: wait for element that never appears (timeout)\n>       expect(page.get_by_text(\"This will never appear\")).to_be_visible(timeout=3000)\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 3000ms\nE         -   - waiting for get_by_text(\"This will never appear\")\n\ntests/test_example.py:40: AssertionError",
        "start_time": 1754996137.3829708,
        "status": "failed",
        "thread": "34136-MainThread"
    },
    "fields": {},
    "id": "fb8d6919-87ac-4647-ba35-9703a0ec2571",
    "message": "AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 3000ms\n  -   - waiting for get_by_text(\"This will never appear\")",
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_fail_timeout[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_fail_timeout"
}
[Qase][16:25:40][info] See why this test failed: https://app.qase.io/run/RANDOM/dashboard/76?source=logs&status=%5B2%5D&search=test_fail_timeout
[Qase][16:25:40][debug] Result test_fail_timeout added
[Qase][16:25:46][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 5575,
        "end_time": 1754996146.495163,
        "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_incorrect_attribute(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: assert attribute that is wrong\n        elem = page.get_by_role(\"link\", name=\"Get started\")\n>       expect(elem).to_have_attribute(\"href\", \"/wrong-link\")\nE       AssertionError: Locator expected to have attribute '/wrong-link'\nE       Actual value: /docs/intro \nE       Call log:\nE         - LocatorAssertions.to_have_attribute with timeout 5000ms\nE         -   - waiting for get_by_role(\"link\", name=\"Get started\")\nE         -     9 \u00d7 locator resolved to <a href=\"/docs/intro\" class=\"getStarted_Sjon\">Get started</a>\nE         -       - unexpected value \"/docs/intro\"\n\ntests/test_example.py:47: AssertionError",
        "start_time": 1754996140.919662,
        "status": "failed",
        "thread": "34136-MainThread"
    },
    "fields": {},
    "id": "d7a73910-700e-4452-a897-5157db24d930",
    "message": "AssertionError: Locator expected to have attribute '/wrong-link'\nActual value: /docs/intro \nCall log:\n  - LocatorAssertions.to_have_attribute with timeout 5000ms\n  -   - waiting for get_by_role(\"link\", name=\"Get started\")\n  -     9 \u00d7 locator resolved to <a href=\"/docs/intro\" class=\"getStarted_Sjon\">Get started</a>\n  -       - unexpected value \"/docs/intro\"",
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_fail_incorrect_attribute[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_fail_incorrect_attribute"
}
[Qase][16:25:46][info] See why this test failed: https://app.qase.io/run/RANDOM/dashboard/76?source=logs&status=%5B2%5D&search=test_fail_incorrect_attribute
[Qase][16:25:46][debug] Result test_fail_incorrect_attribute added
[Qase][16:25:46][debug] Prepared result: {"title": "test_has_title", "signature": "tests::test_example.py::test_has_title[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996112.6774888, "end_time": 1754996115.133686, "status": "passed", "duration": 2456, "thread": "34136-MainThread", "stacktrace": null}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "defect": false, "testops_ids": null, "message": null}
[Qase][16:25:46][debug] Prepared result: {"title": "test_get_started_link", "signature": "tests::test_example.py::test_get_started_link[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996115.13417, "end_time": 1754996120.756883, "status": "failed", "duration": 5622, "stacktrace": "page = <Page url='https://playwright.dev/docs/intro'>\n\n    def test_get_started_link(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: typo in heading name\n        page.get_by_role(\"link\", name=\"Get started\").click()\n>       expect(page.get_by_role(\"heading\", name=\"Instalation\")).to_be_visible()  # typo\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 5000ms\nE         -   - waiting for get_by_role(\"heading\", name=\"Instalation\")\n\ntests/test_example.py:15: AssertionError", "thread": "34136-MainThread"}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "message": "AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 5000ms\n  -   - waiting for get_by_role(\"heading\", name=\"Instalation\")", "defect": false, "testops_ids": null}
[Qase][16:25:46][debug] Prepared result: {"title": "test_fail_wrong_url", "signature": "tests::test_example.py::test_fail_wrong_url[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996120.7584739, "end_time": 1754996126.2565382, "status": "failed", "duration": 5498, "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_wrong_url(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: wrong URL assertion\n>       expect(page).to_have_url(\"https://example.com\")  # wrong URL\nE       AssertionError: Page URL expected to be 'https://example.com'\nE       Actual value: https://playwright.dev/ \nE       Call log:\nE         - PageAssertions.to_have_url with timeout 5000ms\nE         -   - waiting for locator(\":root\")\nE         -     9 \u00d7 locator resolved to <html lang=\"en\" dir=\"ltr\" data-theme=\"light\" data-has-hydrated=\"true\" data-theme-choice=\"system\" class=\"plugin-pages plugin-id-default\" data-rh=\"lang,dir,class,data-has-hydrated\">\u2026</html>\nE         -       - unexpected value \"https://playwright.dev/\"\n\ntests/test_example.py:21: AssertionError", "thread": "34136-MainThread"}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "message": "AssertionError: Page URL expected to be 'https://example.com'\nActual value: https://playwright.dev/ \nCall log:\n  - PageAssertions.to_have_url with timeout 5000ms\n  -   - waiting for locator(\":root\")\n  -     9 \u00d7 locator resolved to <html lang=\"en\" dir=\"ltr\" data-theme=\"light\" data-has-hydrated=\"true\" data-theme-choice=\"system\" class=\"plugin-pages plugin-id-default\" data-rh=\"lang,dir,class,data-has-hydrated\">\u2026</html>\n  -       - unexpected value \"https://playwright.dev/\"", "defect": false, "testops_ids": null}
[Qase][16:25:46][debug] Prepared result: {"title": "test_fail_element_not_found", "signature": "tests::test_example.py::test_fail_element_not_found[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996126.2573519, "end_time": 1754996131.806358, "status": "failed", "duration": 5549, "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_element_not_found(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: element that doesn't exist\n>       expect(page.get_by_text(\"Nonexistent Text\")).to_be_visible()\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 5000ms\nE         -   - waiting for get_by_text(\"Nonexistent Text\")\n\ntests/test_example.py:27: AssertionError", "thread": "34136-MainThread"}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "message": "AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 5000ms\n  -   - waiting for get_by_text(\"Nonexistent Text\")", "defect": false, "testops_ids": null}
[Qase][16:25:46][debug] Prepared result: {"title": "test_fail_incorrect_text", "signature": "tests::test_example.py::test_fail_incorrect_text[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996131.807461, "end_time": 1754996137.3817348, "status": "failed", "duration": 5574, "stacktrace": "page = <Page url='https://playwright.dev/docs/intro'>\n\n    def test_fail_incorrect_text(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        page.get_by_role(\"link\", name=\"Get started\").click()\n        # Fail: text is not matching\n>       expect(page.get_by_role(\"heading\", name=\"Installation\")).to_have_text(\"Wrong Text\")\nE       AssertionError: Locator expected to have text 'Wrong Text'\nE       Actual value: Installation \nE       Call log:\nE         - LocatorAssertions.to_have_text with timeout 5000ms\nE         -   - waiting for get_by_role(\"heading\", name=\"Installation\")\nE         -     6 \u00d7 locator resolved to <h1>Installation</h1>\nE         -       - unexpected value \"Installation\"\n\ntests/test_example.py:34: AssertionError", "thread": "34136-MainThread"}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "message": "AssertionError: Locator expected to have text 'Wrong Text'\nActual value: Installation \nCall log:\n  - LocatorAssertions.to_have_text with timeout 5000ms\n  -   - waiting for get_by_role(\"heading\", name=\"Installation\")\n  -     6 \u00d7 locator resolved to <h1>Installation</h1>\n  -       - unexpected value \"Installation\"", "defect": false, "testops_ids": null}
[Qase][16:25:46][debug] Prepared result: {"title": "test_fail_timeout", "signature": "tests::test_example.py::test_fail_timeout[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996137.3829708, "end_time": 1754996140.9187148, "status": "failed", "duration": 3535, "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_timeout(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: wait for element that never appears (timeout)\n>       expect(page.get_by_text(\"This will never appear\")).to_be_visible(timeout=3000)\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 3000ms\nE         -   - waiting for get_by_text(\"This will never appear\")\n\ntests/test_example.py:40: AssertionError", "thread": "34136-MainThread"}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "message": "AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 3000ms\n  -   - waiting for get_by_text(\"This will never appear\")", "defect": false, "testops_ids": null}
[Qase][16:25:46][debug] Prepared result: {"title": "test_fail_incorrect_attribute", "signature": "tests::test_example.py::test_fail_incorrect_attribute[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996140.919662, "end_time": 1754996146.495163, "status": "failed", "duration": 5575, "stacktrace": "page = <Page url='https://playwright.dev/'>\n\n    def test_fail_incorrect_attribute(page: Page):\n        page.goto(\"https://playwright.dev/\")\n        # Fail: assert attribute that is wrong\n        elem = page.get_by_role(\"link\", name=\"Get started\")\n>       expect(elem).to_have_attribute(\"href\", \"/wrong-link\")\nE       AssertionError: Locator expected to have attribute '/wrong-link'\nE       Actual value: /docs/intro \nE       Call log:\nE         - LocatorAssertions.to_have_attribute with timeout 5000ms\nE         -   - waiting for get_by_role(\"link\", name=\"Get started\")\nE         -     9 \u00d7 locator resolved to <a href=\"/docs/intro\" class=\"getStarted_Sjon\">Get started</a>\nE         -       - unexpected value \"/docs/intro\"\n\ntests/test_example.py:47: AssertionError", "thread": "34136-MainThread"}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "message": "AssertionError: Locator expected to have attribute '/wrong-link'\nActual value: /docs/intro \nCall log:\n  - LocatorAssertions.to_have_attribute with timeout 5000ms\n  -   - waiting for get_by_role(\"link\", name=\"Get started\")\n  -     9 \u00d7 locator resolved to <a href=\"/docs/intro\" class=\"getStarted_Sjon\">Get started</a>\n  -       - unexpected value \"/docs/intro\"", "defect": false, "testops_ids": null}
[Qase][16:25:46][debug] Sending results for run 76: [ResultCreate(id=None, title='test_has_title', signature='tests::test_example.py::test_has_title[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996112.6774888, end_time=1754996115.133686, status='passed', duration=2456, stacktrace=None, thread='34136-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message=None, defect=False), ResultCreate(id=None, title='test_get_started_link', signature='tests::test_example.py::test_get_started_link[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996115.13417, end_time=1754996120.756883, status='failed', duration=5622, stacktrace='page = <Page url=\'https://playwright.dev/docs/intro\'>\n\n    def test_get_started_link(page: Page):\n        page.goto("https://playwright.dev/")\n        # Fail: typo in heading name\n        page.get_by_role("link", name="Get started").click()\n>       expect(page.get_by_role("heading", name="Instalation")).to_be_visible()  # typo\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 5000ms\nE         -   - waiting for get_by_role("heading", name="Instalation")\n\ntests/test_example.py:15: AssertionError', thread='34136-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message='AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 5000ms\n  -   - waiting for get_by_role("heading", name="Instalation")', defect=False), ResultCreate(id=None, title='test_fail_wrong_url', signature='tests::test_example.py::test_fail_wrong_url[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996120.7584739, end_time=1754996126.2565382, status='failed', duration=5498, stacktrace='page = <Page url=\'https://playwright.dev/\'>\n\n    def test_fail_wrong_url(page: Page):\n        page.goto("https://playwright.dev/")\n        # Fail: wrong URL assertion\n>       expect(page).to_have_url("https://example.com")  # wrong URL\nE       AssertionError: Page URL expected to be \'https://example.com\'\nE       Actual value: https://playwright.dev/ \nE       Call log:\nE         - PageAssertions.to_have_url with timeout 5000ms\nE         -   - waiting for locator(":root")\nE         -     9 × locator resolved to <html lang="en" dir="ltr" data-theme="light" data-has-hydrated="true" data-theme-choice="system" class="plugin-pages plugin-id-default" data-rh="lang,dir,class,data-has-hydrated">…</html>\nE         -       - unexpected value "https://playwright.dev/"\n\ntests/test_example.py:21: AssertionError', thread='34136-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message='AssertionError: Page URL expected to be \'https://example.com\'\nActual value: https://playwright.dev/ \nCall log:\n  - PageAssertions.to_have_url with timeout 5000ms\n  -   - waiting for locator(":root")\n  -     9 × locator resolved to <html lang="en" dir="ltr" data-theme="light" data-has-hydrated="true" data-theme-choice="system" class="plugin-pages plugin-id-default" data-rh="lang,dir,class,data-has-hydrated">…</html>\n  -       - unexpected value "https://playwright.dev/"', defect=False), ResultCreate(id=None, title='test_fail_element_not_found', signature='tests::test_example.py::test_fail_element_not_found[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996126.2573519, end_time=1754996131.806358, status='failed', duration=5549, stacktrace='page = <Page url=\'https://playwright.dev/\'>\n\n    def test_fail_element_not_found(page: Page):\n        page.goto("https://playwright.dev/")\n        # Fail: element that doesn\'t exist\n>       expect(page.get_by_text("Nonexistent Text")).to_be_visible()\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 5000ms\nE         -   - waiting for get_by_text("Nonexistent Text")\n\ntests/test_example.py:27: AssertionError', thread='34136-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message='AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 5000ms\n  -   - waiting for get_by_text("Nonexistent Text")', defect=False), ResultCreate(id=None, title='test_fail_incorrect_text', signature='tests::test_example.py::test_fail_incorrect_text[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996131.807461, end_time=1754996137.3817348, status='failed', duration=5574, stacktrace='page = <Page url=\'https://playwright.dev/docs/intro\'>\n\n    def test_fail_incorrect_text(page: Page):\n        page.goto("https://playwright.dev/")\n        page.get_by_role("link", name="Get started").click()\n        # Fail: text is not matching\n>       expect(page.get_by_role("heading", name="Installation")).to_have_text("Wrong Text")\nE       AssertionError: Locator expected to have text \'Wrong Text\'\nE       Actual value: Installation \nE       Call log:\nE         - LocatorAssertions.to_have_text with timeout 5000ms\nE         -   - waiting for get_by_role("heading", name="Installation")\nE         -     6 × locator resolved to <h1>Installation</h1>\nE         -       - unexpected value "Installation"\n\ntests/test_example.py:34: AssertionError', thread='34136-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message='AssertionError: Locator expected to have text \'Wrong Text\'\nActual value: Installation \nCall log:\n  - LocatorAssertions.to_have_text with timeout 5000ms\n  -   - waiting for get_by_role("heading", name="Installation")\n  -     6 × locator resolved to <h1>Installation</h1>\n  -       - unexpected value "Installation"', defect=False), ResultCreate(id=None, title='test_fail_timeout', signature='tests::test_example.py::test_fail_timeout[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996137.3829708, end_time=1754996140.9187148, status='failed', duration=3535, stacktrace='page = <Page url=\'https://playwright.dev/\'>\n\n    def test_fail_timeout(page: Page):\n        page.goto("https://playwright.dev/")\n        # Fail: wait for element that never appears (timeout)\n>       expect(page.get_by_text("This will never appear")).to_be_visible(timeout=3000)\nE       AssertionError: Locator expected to be visible\nE       Actual value: <element(s) not found> \nE       Call log:\nE         - LocatorAssertions.to_be_visible with timeout 3000ms\nE         -   - waiting for get_by_text("This will never appear")\n\ntests/test_example.py:40: AssertionError', thread='34136-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message='AssertionError: Locator expected to be visible\nActual value: <element(s) not found> \nCall log:\n  - LocatorAssertions.to_be_visible with timeout 3000ms\n  -   - waiting for get_by_text("This will never appear")', defect=False), ResultCreate(id=None, title='test_fail_incorrect_attribute', signature='tests::test_example.py::test_fail_incorrect_attribute[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996140.919662, end_time=1754996146.495163, status='failed', duration=5575, stacktrace='page = <Page url=\'https://playwright.dev/\'>\n\n    def test_fail_incorrect_attribute(page: Page):\n        page.goto("https://playwright.dev/")\n        # Fail: assert attribute that is wrong\n        elem = page.get_by_role("link", name="Get started")\n>       expect(elem).to_have_attribute("href", "/wrong-link")\nE       AssertionError: Locator expected to have attribute \'/wrong-link\'\nE       Actual value: /docs/intro \nE       Call log:\nE         - LocatorAssertions.to_have_attribute with timeout 5000ms\nE         -   - waiting for get_by_role("link", name="Get started")\nE         -     9 × locator resolved to <a href="/docs/intro" class="getStarted_Sjon">Get started</a>\nE         -       - unexpected value "/docs/intro"\n\ntests/test_example.py:47: AssertionError', thread='34136-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message='AssertionError: Locator expected to have attribute \'/wrong-link\'\nActual value: /docs/intro \nCall log:\n  - LocatorAssertions.to_have_attribute with timeout 5000ms\n  -   - waiting for get_by_role("link", name="Get started")\n  -     9 × locator resolved to <a href="/docs/intro" class="getStarted_Sjon">Get started</a>\n  -       - unexpected value "/docs/intro"', defect=False)]
[Qase][16:25:47][debug] Results for run 76 sent successfully
[Qase][16:25:47][debug] Completing run
[Qase][16:25:47][debug] Completing run 76
[Qase][16:25:48][info] Test run link: https://app.qase.io/run/RANDOM/dashboard/76
[Qase][16:25:48][debug] Run completed
[Qase][16:25:48][info] Overhead for Qase Report: 2938ms
[Qase][16:28:12][debug] Config: {
    "debug": true,
    "environment": "local",
    "exclude_params": [],
    "execution_plan": {
        "path": "./build/qase-execution-plan.json"
    },
    "fallback": "report",
    "framework": {
        "playwright": {
            "output_dir": "test-results",
            "trace": "off",
            "video": "off"
        },
        "pytest": {
            "capture_logs": true,
            "xfail_status": {
                "xfail": "skipped",
                "xpass": "passed"
            }
        }
    },
    "mode": "testops",
    "profilers": [
        "network",
        "sleep"
    ],
    "report": {
        "connection": {
            "format": "json",
            "path": "./build/qase-report"
        },
        "driver": "local"
    },
    "testops": {
        "api": {
            "host": "qase.io",
            "token": "bf98087c2b769f39a4741acf720a27171d3128ec64fa2a5bc4987e7425834825"
        },
        "batch": {
            "size": 100
        },
        "defect": false,
        "plan": {},
        "project": "RANDOM",
        "run": {
            "complete": true,
            "tags": [],
            "title": "pytestrun"
        }
    }
}
[Qase][16:28:13][debug] Host data: {'system': 'darwin', 'machineName': 'Ankitas-MacBook-Air.local', 'release': '24.6.0', 'version': '15.6', 'arch': 'arm64', 'python': '3.13.5', 'pip': '24.3.1', 'framework': '8.3.4', 'reporter': '6.3.3', 'commons': '3.5.0', 'apiClientV1': '1.2.1', 'apiClientV2': '1.2.1'}
[Qase][16:28:13][debug] Preparing API client
[Qase][16:28:13][debug] API client prepared
[Qase][16:28:13][debug] Preparing API V2 client
[Qase][16:28:13][debug] API V2 client prepared
[Qase][16:28:13][debug] Getting environment local
[Qase][16:28:15][debug] Environment local not found
[Qase][16:28:15][debug] Getting project RANDOM
[Qase][16:28:16][debug] Project RANDOM found: {"title": "Random", "code": "RANDOM", "counts": {"cases": 14, "suites": 7, "milestones": 0, "runs": {"total": 48, "active": 14}, "defects": {"total": 3, "open": 3}}}
[Qase][16:28:16][debug] Starting run
[Qase][16:28:16][debug] Creating test run with parameters: {'title': 'pytestrun', 'description': 'Automated Run 2025-08-12 16:28:15.645133', 'environment_id': None, 'plan_id': None, 'is_autotest': True, 'start_time': '2025-08-12 10:58:16', 'tags': []}
[Qase][16:28:17][debug] Run ID: 77
[Qase][16:28:19][debug] Adding result {
    "attachments": [],
    "execution": {
        "duration": 1302,
        "end_time": 1754996299.218175,
        "stacktrace": null,
        "start_time": 1754996297.915529,
        "status": "passed",
        "thread": "34290-MainThread"
    },
    "fields": {},
    "id": "fedca76a-53c0-484a-98fa-62a59cd89775",
    "message": null,
    "muted": false,
    "param_groups": [],
    "params": {
        "browser_name": "chromium"
    },
    "relations": {
        "suite": {
            "data": [
                {
                    "public_id": null,
                    "title": "tests"
                },
                {
                    "public_id": null,
                    "title": "test_example"
                }
            ]
        }
    },
    "signature": "tests::test_example.py::test_has_title[chromium]::{browser_name:chromium}",
    "steps": [],
    "testops_ids": null,
    "title": "test_has_title"
}
[Qase][16:28:19][debug] Result test_has_title added
[Qase][16:28:19][debug] Prepared result: {"title": "test_has_title", "signature": "tests::test_example.py::test_has_title[chromium]::{browser_name:chromium}", "execution": {"start_time": 1754996297.915529, "end_time": 1754996299.218175, "status": "passed", "duration": 1302, "thread": "34290-MainThread", "stacktrace": null}, "fields": {}, "attachments": [], "steps": [], "steps_type": "classic", "params": {"browser_name": "chromium"}, "param_groups": [], "relations": {"suite": {"data": [{"title": "tests"}, {"title": "test_example"}]}}, "defect": false, "testops_ids": null, "message": null}
[Qase][16:28:19][debug] Sending results for run 77: [ResultCreate(id=None, title='test_has_title', signature='tests::test_example.py::test_has_title[chromium]::{browser_name:chromium}', testops_id=None, testops_ids=None, execution=ResultExecution(start_time=1754996297.915529, end_time=1754996299.218175, status='passed', duration=1302, stacktrace=None, thread='34290-MainThread'), fields=ResultCreateFields(author=None, description=None, preconditions=None, postconditions=None, layer=None, severity=None, priority=None, behavior=None, type=None, muted=None, is_flaky=None, executed_by=None, additional_properties={}), attachments=[], steps=[], steps_type=<ResultStepsType.CLASSIC: 'classic'>, params={'browser_name': 'chromium'}, param_groups=[], relations=ResultRelations(suite=RelationSuite(data=[RelationSuiteItem(title='tests', public_id=None), RelationSuiteItem(title='test_example', public_id=None)])), message=None, defect=False)]
[Qase][16:28:20][debug] Results for run 77 sent successfully
[Qase][16:28:20][debug] Completing run
[Qase][16:28:20][debug] Completing run 77
[Qase][16:28:21][info] Test run link: https://app.qase.io/run/RANDOM/dashboard/77
[Qase][16:28:21][debug] Run completed
[Qase][16:28:21][info] Overhead for Qase Report: 2757ms
